extra_foam/gui/image_tool/azimuthal_integ_1d_view.py

```diff
@@ -97,9 +100,20 @@ def initUI(self):
         layout.setContentsMargins(1, 1, 1, 1)
         self.setLayout(layout)
 
+        self._corrected._plot_widget.addItem(self._crosshair)
+
     def initConnections(self):
         """Override."""
-        pass
+        self._ctrl_widget.cx_changed_sgn.connect(lambda x: self.onBeamCenterChanged(x, None))
+        self._ctrl_widget.cy_changed_sgn.connect(lambda y: self.onBeamCenterChanged(None, y))
+
+    def onBeamCenterChanged(self, x, y):
+        if x is None:
+            x = float(self._ctrl_widget._cx_le.text())
```

Comment:
You could use `lineedit.value()` to avoid casting it to float. Seems that the smart edits casts to the appropriate type when a validator is added

Reply:
Unfortunately that only works with some subclasses of `SmartLineEdit`, which have special `Validator` inner classes that `value()` uses to do the casting.